src/mish2jsonl/mish2jsonl.c

```diff
@@ -0,0 +1,160 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include "mish_common.h"
+
+typedef struct m_string {
+  uint64_t len;
+  char *string;
+} m_string;
+
+typedef struct worker_output {
+  uint32_t status;   // 4
+  uint16_t ndecoded; // 2
+  uint32_t workerno; // 4
+  m_string workerso; // 16
+  m_string result;   // 16
+} worker_output;
+
+typedef struct cohort_results {
+  uint32_t nworkers;
+  m_string input;
+  worker_output *outputs;
+} cohort_results;
+
+cohort_results results;
+FILE *m_parsing_file;
+int m_finished_parsing;
+
+void m_cohort_print_json(FILE *f, cohort_results *r) {
+  fprintf(f, "{ \"nworkers\": %u, \"input\": \"%s\", \"outputs\": [", r->nworkers, r->input.string);
+  for (int i = 0; i < r->nworkers; i++) {
+    if (i != 0) {
+      fprintf(f, ",");
+    }
+    fprintf(f,
+            "{ \"status\": { \"value\": %u, \"name\": \"%s\" }, \"ndecoded\": %u, \"workerno\": "
+            "%u, \"worker_so\": \"%s\",\"len\": %ld, \"result\": \"%s\" }",
+            r->outputs[i].status, status2str(r->outputs[i].status), r->outputs[i].ndecoded,
+            r->outputs[i].workerno, r->outputs[i].workerso.string, r->outputs[i].result.len,
+            r->outputs[i].status == 1 ? r->outputs[i].result.string : "bad");
+  }
+
+  fprintf(f, "]}");
+  return;
+}
+
+void m_fread(void *ref, size_t size, size_t times, FILE *file) {
+  size_t rd = fread(ref, size, times, file);
+  if (rd == 0 &&
+      size * times != 0) { // in cases of bad decompilation we want to read 0 length result string
+    m_finished_parsing = 1;
+  }
+}
+
+/*
+    There are a few subtleties that this functions catches
+    1) Not all strings we will read actually have an null byte
+    even when we read the entire string, leading to extra
+    junk being inserted in for example an output file
+    and hence crashing the analysis tool
+
+    2) Some disassemblers like to insert \n in their output
+    this breaks values from being a valid string
+
+    3) at the moment we alloc the memory of the string here
+    using malloc will result in padded whitespaces showing
+    old data that we don't want. Hence the use of calloc
+    if this is only used for printing one-by-one
+    we can optimize this out
+
+    we implicitly calloc 1 extra byte to guarantee
+    the string ends with a null byte (implicit calloc logic)
+*/
+
+void read_string(FILE *file, m_string *s, int len_size) {
+  uint64_t string_length = 0;
+  m_fread(&string_length, len_size, 1, file);
+
+  char *input =
+      calloc(1, sizeof(char) * string_length +
+                    1); // calloc instead of malloc because we want to zero out the memory.
```

Comment:
`clang-format` mangled these lines because of the inline comment -- maybe move the comment to its own line so that the formatting is a little more reasonable?